libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java

```diff
@@ -438,6 +488,26 @@ public void run() {
         return view;
     }
 
+    private String calculateHashOnMediaCollectionBasedBlock(ArrayList<Object> mediaFiles) {
+        Gson gson = new Gson();
+        // make sure to normalize ids to Strings as these may vary and make the hash not coincide
+        normalizeMediaFilesIds(mediaFiles);
+        return StringUtils.getMd5Hash(gson.toJson(mediaFiles));
+    }
+
+    private void normalizeMediaFilesIds(ArrayList<Object> mediaFiles) {
+        // iterate through all of mediaFiles objects    and convert ids to String
+        for (Object mediaFile : mediaFiles) {
+            // this conversion is needed to strip off decimals that can come from RN when using int as
+            // string
+            if (((HashMap<String, Object>) mediaFile).get("id") instanceof Double) {
+                Double originalValue = (Double) ((HashMap<String, Object>) mediaFile).get("id");
+                // now set it back to String to normalize with temporary ids
+                ((HashMap) mediaFile).put("id", String.valueOf(originalValue.longValue()));
```

Comment:
@mzorz I would normally hardcode `id` value into a constant but that's totally not required. I just spotted it and decided to share 😜

Reply:
I totally get what you say and I'd definitely go for a constant when code is self-contained - in this situation I personally find it easier to read the `"id"` field right there on the spot given it is shared with the JavaScript counterpart so, when following code I have one less level of abstraction / indirection to check (as I can spot the `"id"` key right away when looking at both codebases, as opposed to checking a constant, figuring out what the constant value is, then trying to mentally match it to what the JS counterpart is doing). For this reason I'd prefer to leave it, but happy to change it otherwise, what do you think @jd-alexander ?

Reply:
Yes it can remain. Now when I think about the main reason my mind told me to put it in a constant would be if we needed to replace the value we would only make this change in one place but now I can't think of any reason where id itself would be changed since it's a universal way of identifying objects so in that sense, for readability and overall consistent it can remain. Thanks for expounding @mzorz